kasa/effects.py

```diff
@@ -0,0 +1,296 @@
+    "backgrounds": [[200, 100, 100], [200, 50, 10], [210, 10, 50], [240, 10, 0]],
+}
+EFFECT_OCEAN = {
+    "custom": 0,
+    "id": "oJjUMosgEMrdumfPANKbkFmBcAdEQsPy",
+    "brightness": 30,
+    "name": "Ocean",
+    "segments": [0],
+    "expansion_strategy": 1,
+    "enable": 1,
+    "type": "sequence",
+    "duration": 0,
+    "transition": 2000,
+    "direction": 3,
+    "spread": 16,
+    "repeat_times": 0,
+    "sequence": [[198, 84, 30], [198, 70, 30], [198, 10, 30]],
+}
+EFFECT_RAINBOW = {
+    "custom": 0,
+    "id": "izRhLCQNcDzIKdpMPqSTtBMuAIoreAuT",
+    "brightness": 100,
+    "name": "Rainbow",
+    "segments": [0],
+    "expansion_strategy": 1,
+    "enable": 1,
+    "type": "sequence",
+    "duration": 0,
+    "transition": 1500,
+    "direction": 1,
+    "spread": 12,
+    "repeat_times": 0,
+    "sequence": [[0, 100, 100], [100, 100, 100], [200, 100, 100], [300, 100, 100]],
+}
+EFFECT_RAINDROP = {
+    "custom": 0,
+    "id": "QbDFwiSFmLzQenUOPnJrsGqyIVrJrRsl",
+    "brightness": 30,
+    "name": "Raindrop",
+    "segments": [0],
+    "expansion_strategy": 1,
+    "enable": 1,
+    "type": "random",
+    "hue_range": [200, 200],
+    "saturation_range": [10, 20],
+    "brightness_range": [10, 30],
+    "duration": 0,
+    "transition": 1000,
+    "init_states": [[200, 40, 100]],
+    "fadeoff": 1000,
+    "random_seed": 24,
+    "backgrounds": [[200, 40, 0]],
+}
+EFFECT_SPRING = {
+    "custom": 0,
+    "id": "URdUpEdQbnOOechDBPMkKrwhSupLyvAg",
+    "brightness": 100,
+    "name": "Spring",
+    "segments": [0],
+    "expansion_strategy": 1,
+    "enable": 1,
+    "type": "random",
+    "hue_range": [0, 90],
+    "saturation_range": [30, 100],
+    "brightness_range": [90, 100],
+    "duration": 600,
+    "transition": 0,
+    "transition_range": [2000, 6000],
+    "init_states": [[80, 30, 100]],
+    "fadeoff": 1000,
+    "random_seed": 20,
+    "backgrounds": [[130, 100, 40]],
+}
+EFFECT_VALENTINES = {
+    "custom": 0,
+    "id": "QglBhMShPHUAuxLqzNEefFrGiJwahOmz",
+    "brightness": 100,
+    "name": "Valentines",
+    "segments": [0],
+    "expansion_strategy": 1,
+    "enable": 1,
+    "type": "random",
+    "hue_range": [340, 340],
+    "saturation_range": [30, 40],
+    "brightness_range": [90, 100],
+    "duration": 600,
+    "transition": 2000,
+    "init_states": [[340, 30, 100]],
+    "fadeoff": 3000,
+    "random_seed": 100,
+    "backgrounds": [[340, 20, 50], [20, 50, 50], [0, 100, 50]],
+}
+
+EFFECTS_LIST_V1 = [
+    EFFECT_AURORA,
+    EFFECT_BUBBLING_CAULDRON,
+    EFFECT_CANDY_CANE,
+    EFFECT_CHRISTMAS,
+    EFFECT_FLICKER,
+    EFFECT_HANUKKAH,
+    EFFECT_HAUNTED_MANSION,
+    EFFECT_ICICLE,
+    EFFECT_LIGHTNING,
+    EFFECT_OCEAN,
+    EFFECT_RAINBOW,
+    EFFECT_RAINDROP,
+    EFFECT_SPRING,
+    EFFECT_VALENTINES,
+]
+
+EFFECT_NAMES_V1: List[str] = [cast(str, effect["name"]) for effect in EFFECTS_LIST_V1]
+EFFECT_MAPPING_V1 = {effect["name"]: effect for effect in EFFECTS_LIST_V1}
```

Comment:
```suggestion
EFFECT_MAPPING_V1 = {effect["name"]: effect for effect in EFFECTS_LIST_V1}
```
No need for a separate list for names, `if name in EFFECT_MAPPING_V1` works fine for checking the existence.

Reply:
I'm using the list for the `effect_list` property

Reply:
In that case, `list(effect_list)` would give a list :-)

Reply:
mypy doesn't seem to like that:

`kasa/smartlightstrip.py:76: error: Argument 1 to "list" has incompatible type "Dict[object, Dict[str, object]]"; expected "Iterable[str]"`

Reply:
Oh, well, `.keys()` should work definitely though.

Reply:
It had to be cast since the dict is mixed and has values of `Any`